### PR TITLE
feat(FeatureGoogleAnalytics): Removed unused options and modified to …

### PR DESCRIPTION
**Pull request by MyEmpireOfDuctTape**
…fit opt in

### CODE REVIEW

File: Components/BlockCookieNotice/functions.php [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 
 Options::addGlobal('BlockCookieNotice', [
+    [
+        'label' => 'Cookie notice is enabled',
+        'name' => 'cookieNoticeIsEnabled',
+        'type' => 'true_false',
+        'default_value' => 1,
+        'ui' => 1,
+        'ui_on_text' => 'Is Enabled',
+        'ui_off_text' => 'Isn\'t Enabled',
```
Comment (steffenbew) [inline Components/BlockCookieNotice/functions.php]: I would like to propose different label and values:
Label: "Component Status"
Values: "Activated / Deactivated"

File: Components/FeatureGoogleAnalytics/README.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
 
-If a Google Analytics ID is provided in Global Options, the tracking code will be added to all pages, expect for logged in WordPress admins. Entering "debug" instead will log to the browser console only.
+Enable the Google Analytics tracking configuration in the WordPress back end.
+
+If a valid Google Analytics ID is provided, the tracking code will automatically be added to all pages.
+
+The following options are provided under the **"Global Options"** menu:
+
+- Google Analytics ID: This must be a valid Google Analytics ID.
+- Anonymize IP: Check to enable the Google Analytics `_anonymizeIp` function and anonymize the last digits of the user's IP.
```
Comment (steffenbew) [inline Components/FeatureGoogleAnalytics/README.md]: Please keep the shorter readme (I just updated it 2 days ago). But could you please fix my typo: "expect -> except"?

File: Components/FeatureGoogleAnalytics/functions.php [hunk 3]
```diff
@@ -2,21 +2,63 @@
...
-{
+add_filter('Flynt/addComponentData?name=FeatureGoogleAnalytics', function ($data) {
     $googleAnalyticsOptions = Options::getGlobal('GoogleAnalytics');
+
     if ($googleAnalyticsOptions) {
-        new GoogleAnalytics($googleAnalyticsOptions);
+        $data['jsonData'] = json_encode([
+            'gaId' => $googleAnalyticsOptions['gaId'],
+            'anonymizeIp' => $googleAnalyticsOptions['anonymizeIp'],
+            'serverSideTrackingEnabled' => isTrackingEnabled($googleAnalyticsOptions['gaId']),
```
Comment (steffenbew) [inline Components/FeatureGoogleAnalytics/functions.php]: How about using "isTrackingEnabled' as a variable name, according to the function name? While looking into that function, I'm wondering if you removed the debug mode?

Comment (MyEmpireOfDuctTape) [reply Components/FeatureGoogleAnalytics/functions.php]: I've changed the variable to be called 'isTrackingEnabled' now. I also removed the debug mode from the function. Was this not necessary?

Comment (steffenbew) [reply Components/FeatureGoogleAnalytics/functions.php]: The debug mode should still be there.

File: Components/BlockCookieNotice/functions.php (hunk 1 above)
Comment (MyEmpireOfDuctTape) [reply Components/BlockCookieNotice/functions.php]: I changed the label values as specified.

File: Components/FeatureGoogleAnalytics/README.md (hunk 2 above)
Comment (MyEmpireOfDuctTape) [reply Components/FeatureGoogleAnalytics/README.md]: I added the readme from the previous version and fixed the typo.

File: Components/FeatureGoogleAnalytics/functions.php (hunk 3 above)
Comment (MyEmpireOfDuctTape) [reply Components/FeatureGoogleAnalytics/functions.php]: I added the debug mode back now. It's only function would be to see what is pushed to the data layer, which it now logs instead, if the GA ID is set to 'debug'

**Comment by MyEmpireOfDuctTape** [comment]
@domtra @steffenbew I've now merged it with the current master which uses the new fieldvariables.

**Pull request merged**